ui/app/components/app/transaction-list/transaction-list-hooks.component.js

```diff
@@ -0,0 +1,161 @@
+import React, { useCallback, useMemo, useEffect, useRef } from 'react'
+import PropTypes from 'prop-types'
+import { useSelector, useDispatch } from 'react-redux'
+import {
+  nonceSortedCompletedTransactionsSelector,
+  nonceSortedPendingTransactionsSelector,
+} from '../../../selectors/transactions'
+import {
+  getSelectedAddress,
+  getAssetImages,
+  getFeatureFlags,
+} from '../../../selectors/selectors'
+import { selectedTokenSelector } from '../../../selectors/tokens'
+import * as actions from '../../../ducks/gas/gas.duck'
+import { useI18nContext } from '../../../hooks/useI18nContext'
+import TransactionListItem from '../transaction-list-item'
+
+
+const mapStateToProps = (state) => {
+  const pendingTransactions = nonceSortedPendingTransactionsSelector(state)
+  const firstPendingTransactionId =
+    pendingTransactions[0] && pendingTransactions[0].primaryTransaction.id
+  return {
+    completedTransactions: nonceSortedCompletedTransactionsSelector(state),
+    pendingTransactions,
+    firstPendingTransactionId,
+    selectedToken: selectedTokenSelector(state),
+    selectedAddress: getSelectedAddress(state),
+    assetImages: getAssetImages(state),
+    transactionTimeFeatureActive: getFeatureFlags(state).transactionTime,
+  }
+}
+
+function useTransactionListState () {
+  const dispatch = useDispatch()
+  const {
+    completedTransactions,
+    firstPendingTransactionId,
+    transactionTimeFeatureActive,
+    assetImages,
+    pendingTransactions,
+    // selectedAddress,
+    selectedToken,
+  } = useSelector(mapStateToProps)
+  const { fetchGasEstimates, fetchBasicGasAndTimeEstimates } = useMemo(() => ({
+    fetchGasEstimates: (blockTime) => dispatch(actions.fetchGasEstimates(blockTime)),
+    fetchBasicGasAndTimeEstimates: () => dispatch(actions.fetchBasicGasAndTimeEstimates()),
+  }), [dispatch])
+  const prevState = useRef({ loaded: false, pendingTransactions, transactionTimeFeatureActive })
+  const { loaded } = prevState.current
+  const pendingTransactionAdded = pendingTransactions.length > 0 && prevState.current.pendingTransactions.length === 0
+  const transactionTimeFeatureWasActivated = !prevState.current.transactionTimeFeatureActive && transactionTimeFeatureActive
+  useEffect(() => {
+    if (transactionTimeFeatureActive && pendingTransactions.length > 0 && (loaded === false || transactionTimeFeatureWasActivated || pendingTransactionAdded)) {
```

Comment:
This single boolean statement should account for both logic in componentDidUpdate and componentDidMount in the component file. loaded is a value we use to see if the component has fired the effect after mount at least once.